File: web/src/types/notificationCluster.ts
// Types matching backend /api/notification_cluster

export type NotificationClusterStatusEnum =
  | 'LEADER'
  | 'FOLLOWER'
  | 'WAITING_FOR_CONSENSUS'
  | 'NOT_PARTICIPATING'

export interface NotificationCluster {
  // key: node_id
  node_statuses: Record<string, NotificationClusterStatusEnum>
}

export interface NotificationClusters {
  // key: cluster name (e.g., channel or integration name)
  clusters: Record<string, NotificationCluster>
}

export interface NotificationClusterApi {
  // key: network id
  networks: Record<string, NotificationClusters>
}

File: src/meshmon/api/structure/notification_cluster.py
from enum import Enum

from pydantic import BaseModel


class NotificationClusterStatusEnum(Enum):
    LEADER = "LEADER"
    FOLLOWER = "FOLLOWER"
    WAITING_FOR_CONSENSUS = "WAITING_FOR_CONSENSUS"
    NOT_PARTICIPATING = "NOT_PARTICIPATING"
    OFFLINE = "OFFLINE"


class NotificationCluster(BaseModel):
    node_statuses: dict[str, NotificationClusterStatusEnum] = {}


class NotificationClusters(BaseModel):
    clusters: dict[str, NotificationCluster] = {}


class NotificationClusterApi(BaseModel):
    networks: dict[str, NotificationClusters] = {}
